hibike/devices/BatteryBuzzer/voltage_sense.cpp

```diff
@@ -0,0 +1,132 @@
+#include "voltage_sense.h"
+#include "pindefs.h"
+
+float v_cell1;  // param 1
+float v_cell2;  // param 2
+float v_cell3;  // param 3
+float v_batt;   // param 4
+float dv_cell2; // param 5
+float dv_cell3; // param 6
+
+//code that deals with measuring cell voltages.
+//also deals with calibration.
+
+void setup_sensing()
+{
+  pinMode(ENABLE, OUTPUT);
+  pinMode(CALIB_BUTTON, INPUT);
+
+  //for stable (if unknown) internal voltage.
+  analogReference(INTERNAL);
+
+  //let's turn enable on, so we can measure the battery voltage.
+  digitalWrite(ENABLE, HIGH);
+
+  if(triple_calibration)
+  {
+    update_triple_calibration();  //and then update the calibration values I use based on what I now know
+  }
+  else
+  {
+    float val = get_calibration();
+    if(val > 0) //by convention, val is -1 if there's no calibration.
+    {
+      vref_guess = val;   
+    }
+  }  
+}
+
+
+void measure_cells() //measures the battery cells. Should call at least twice a second, but preferrably 5x a second.  No use faster.
+{
+    int r_cell1 = analogRead(CELL1);
+    int r_cell2 = analogRead(CELL2);
+    int r_cell3 = analogRead(CELL3);
+
+    v_cell1 = float(r_cell1) * (R2 + R5) / R5 * calib[0] / ADC_COUNTS;  // (10k + 10k)/(10k) to extract cell voltage in counts, then convert 1024 counts per vref volts.
```

Comment:
I think the comments on these lines are no longer necessary, at least with their current content.